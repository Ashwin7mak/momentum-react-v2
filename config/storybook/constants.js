module.exports = {
  REPO_ROOT: '../../../',
  SRC_DIR: './src/',
  ESM_DIR: './dist/esm/',
  CJS_DIR: './dist/cjs/',
  COMPONENTS_DIR: './components/',
};
