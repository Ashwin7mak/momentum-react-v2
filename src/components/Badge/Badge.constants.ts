const CLASS_PREFIX = 'md-badge';

const DEFAULTS = {
  SIZE: 'L',
};

const SIZES = {
  S: 'S',
  M: 'M',
  L: 'L',
};

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
};

export { CLASS_PREFIX, DEFAULTS, SIZES, STYLE };
