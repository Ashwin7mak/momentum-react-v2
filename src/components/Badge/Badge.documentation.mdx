`<Badge />` is an independent component that only utilizes core props.