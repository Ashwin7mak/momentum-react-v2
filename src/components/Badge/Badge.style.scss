.md-badge-wrapper {
    font-family: var(--font-default);
    outline: none;
    border: none;

    background-color: var(--theme-background-accent-normal);
    color: white; //to be fixed when theme token is available

    align-items: center;
    border-radius: 100vh;
    display: flex;

    &[data-size="S"]{
        font-size:0rem;
        height:0.625rem;
        width:0.625rem;
        padding: 0 0;
    }

    &[data-size="M"]{
        font-size:0.5rem;
        line-height: 0.5rem;
        font-weight: medium;
        height:0.75rem;
        padding: 0 0.375rem;
    }

    &[data-size="L"]{
        font-size: 0.75rem;
        line-height: 0.75rem;
        font-weight: medium;
        height: 1.125rem;
        padding: 0 0.375rem;
    }
}
