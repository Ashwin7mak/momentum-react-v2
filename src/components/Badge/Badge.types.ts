import { ReactNode } from 'react';

export interface Props {
  /**
   * Child components of this Badge.
   */
  children?: ReactNode;

  /**
   * Size index of this ButtonPill.
   */
  size?: string;
}
