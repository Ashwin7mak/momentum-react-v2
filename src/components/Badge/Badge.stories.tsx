import React, { FC } from 'react';
import { Story } from '@storybook/react';
import {
  Title,
  Subtitle,
  Description,
  Primary,
  ArgsTable,
  PRIMARY_STORY,
} from '@storybook/addon-docs';

import Badge, { BadgeProps, BADGE_CONSTANTS as CONSTANTS } from './';
import Documentation from './Badge.documentation.mdx';

const DocsPage: FC = () => (
  <>
    <Title />
    <Subtitle />
    <Description />
    <Documentation />
    <Primary />
    <ArgsTable story={PRIMARY_STORY} />
  </>
);

export default {
  title: 'Momentum UI/Badge',
  component: Badge,
  parameters: {
    expanded: true,
    docs: {
      page: DocsPage,
    },
  },
};

const argTypes = {
  children: {
    defaultValue: '10',
    description: 'Provides the child nodes for this element.',
    control: { type: 'text' },
    table: {
      type: {
        summary: 'ReactNode',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
  size: {
    defaultValue: CONSTANTS.DEFAULTS.SIZE,
    description: 'Modifies the size of this `<Badge />`.',
    options: [undefined, ...Object.values(CONSTANTS.SIZES)],
    control: { type: 'select' },
    table: {
      type: {
        summary: 'size',
      },
      defaultValue: {
        summary: CONSTANTS.DEFAULTS.SIZE,
      },
    },
  },
};

const Template: Story<BadgeProps> = (args: BadgeProps) => {
  const mutatedArgs = { ...args };
  delete mutatedArgs.children;

  return <Badge {...mutatedArgs}>{args.children}</Badge>;
};

const MultiTemplate: Story<BadgeProps> = (args: BadgeProps, { parameters }) => {
  const mutatedArgs = { ...args };
  const { children } = mutatedArgs;
  delete mutatedArgs.children;

  const { variants } = parameters;

  const items = variants.map((variant, index: number) => (
    <Badge key={index} {...variant} {...args}>
      {children}
    </Badge>
  ));

  return <>{items}</>;
};

const Example = Template.bind({});

Example.argTypes = { ...argTypes };

const Sizes = MultiTemplate.bind({});

Sizes.parameters = {
  variants: [{}, { size: 'S' }, { size: 'M' }, { size: 'L' }],
};

Sizes.argTypes = { ...argTypes };
delete Sizes.argTypes.size;

export { Example, Sizes };
