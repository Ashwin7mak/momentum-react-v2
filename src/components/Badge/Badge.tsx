import React, { forwardRef, useRef, FC, RefObject } from 'react';

import { DEFAULTS, STYLE } from './Badge.constants';
import { Props } from './Badge.types';
import './Badge.style.scss';

const Badge: FC<Props> = forwardRef((props: Props, providedRef: RefObject<HTMLDivElement>) => {
  const ref = providedRef || useRef();
  const { children } = props;

  return (
    <div className={`${STYLE.wrapper}`} ref={ref} data-size={props.size || DEFAULTS.SIZE}>
      {children}
    </div>
  );
});

Badge.displayName = 'Badge';

export default Badge;
