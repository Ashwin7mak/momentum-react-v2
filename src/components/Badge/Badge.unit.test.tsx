import React from 'react';
import { mount } from 'enzyme';
import Badge from '.';

import { DEFAULTS, SIZES, STYLE } from './Badge.constants';

describe('Badge', () => {
  describe('snapshot', () => {
    let container;

    it('should match snapshot', () => {
      expect.assertions(1);

      container = mount(<Badge>20</Badge>);

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have its main class', () => {
      expect.assertions(1);

      const element = mount(<Badge />)
        .find(Badge)
        .getDOMNode();

      expect(element.classList.contains(STYLE.wrapper)).toBe(true);
    });

    it('should pass child prop', () => {
      expect.assertions(1);

      const child = '20';

      const element = mount(<Badge>{child}</Badge>)
        .find(Badge)
        .childAt(0)
        .getDOMNode();

      expect(element.innerHTML).toBe(child);
    });
  });
});
