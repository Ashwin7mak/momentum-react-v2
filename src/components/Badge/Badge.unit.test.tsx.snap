// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge snapshot should match snapshot 1`] = `
<Badge>
  <div
    className="md-badge-wrapper"
    data-size="L"
  >
    20
  </div>
</Badge>
`;
